### PR TITLE
Improve and clarify cmd help documentation, flags, and examples

### LINKED ISSUE #2422: Go over all flags and document what is optional, what is required, and default values. Capitalize all help messages

### PR DESCRIPTION
**Pull request by scaranoj**
Fixes #2422

### CODE REVIEW

**Comment by scaranoj** [comment]
Closes #2422

**Comment by carlisia** [comment]
@nrb / @ashish-amarnath any idea why we are getting this error for the linter and the CI check:

```
Trying to pull build-image: velero/build-image:c2fd594
docker pull -q velero/build-image:c2fd594 || make build-image
Error response from daemon: pull access denied for velero/build-image, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

**Comment by carlisia** [comment]
https://hub.docker.com/r/velero/velero/tags

**Comment by ashish-amarnath** [comment]
Looking at:
https://github.com/vmware-tanzu/velero/pull/2736/checks?check_run_id=921109303
https://github.com/vmware-tanzu/velero/pull/2736/checks?check_run_id=921109303
Both errors are being caused because `pkg/cmd/cli/version/version.go` was deleted

**Comment by ashish-amarnath** [comment]
@carlisia I am not seeing the docker error you pasted. Can you please link me to where you were seeing those errors?

**Comment by carlisia** [comment]
Here: 

https://github.com/vmware-tanzu/velero/pull/2736/checks?check_run_id=921109303#step:3:107

https://github.com/vmware-tanzu/velero/pull/2736/checks?check_run_id=921109147#step:3:7

File: pkg/cmd/cli/version/version.go [hunk 1]
```diff
@@ -1,81 +0,0 @@
-/*
```
Comment (carlisia) [inline pkg/cmd/cli/version/version.go:1]: @scaranoj this file should not be deleted. Thanks @ashish-amarnath for catching that.

**Comment by ashish-amarnath** [comment]
@carlisia the docker error is because `velero/build-image` is not a valid image. That image is not pushed to docker hub.
The error is coming from https://github.com/vmware-tanzu/velero/blob/main/Makefile#L274

This is basically trying to use a locally cached image, which doesn't exist in CI. Then it attempts to pull down an image from docker hub, and if that doesn't work, then it'll build the `build-image`
Considering that we don't push the build-image to dockerhub, we can remove the `docker pull -q $(BUILDER_IMAGE)` from L274 of the `Makefile`

**Comment by carlisia** [comment]
> @carlisia the docker error is because `velero/build-image` is not a valid image as in that image is not pushed to docker hub.
> The error is coming from https://github.com/vmware-tanzu/velero/blob/main/Makefile#L274
> 
> This is basically trying to use a locally cached image, which doesn't exist in CI. Then it attempts to pull down an image from docker hub, and if that doesn't work, then it'll build the `build-image`
> Considering that we don't push the build-image to dockerhub, we can remove the `docker pull -q $(BUILDER_IMAGE)` from L274 of the `Makefile`

Yes, this make total sense. I can open a PR to remove that. How did we not run into this before?

**Comment by ashish-amarnath** [comment]
We didn't see this locally because we were using cached build-images. In CI, this wouldn't get noticed unless the CI job failed and we didn't know why.

**Comment by nrb** [comment]
@scaranoj Looks like we need a rebase to resolve a conflict with `pkg/cmd/cli/backuplocation/create.go`

**Pull request merged**